hal/src/nRF52840/usart_hal.cpp

```diff
@@ -284,11 +297,18 @@ class Usart {
             // </quote>
             // We'll be extra careful here and make sure not to consume more than we can, otherwise
             // we may put the ring buffer into an invalid state as there are not safety checks.
-            const ssize_t toConsume = std::min(rxBuffer_.acquirePending(), timerValue() - rxConsumed_);
+            size_t timerCount = timerValue();
+            size_t toConsumeCount;
+            if (timerCount >= consumedTimerCount_) {
+                toConsumeCount = timerCount - consumedTimerCount_;
+            } else {
+                toConsumeCount = 65536 - consumedTimerCount_ + timerCount;
+            }
```

Comment:
Just make `timerValue()` return a `uint16_t` here and use `uint16_t` instead of size_t. There are plenty of places that already assume that. Such as setting the width of the timer itself to 16-bit and magic numbers of 65536.

The math breaks if `timerCount` is actually the full width of the underlying `size_t` so I had to read through more code to figure out it was really constrained to 16-bit. Making the types match that constraint makes it more clear and ensures a simple `timerCount - consumedTimerCount_` always presents the correct value without manually adding in 65536.